dyn_em/module_diffusion_em.F

```diff
@@ -5133,7 +5133,7 @@ SUBROUTINE nonlocal_flux (config_flags,nlflux,gamu,gamv,hpbl,kpbl,     &
   DO i = i_start, i_end
       delxy = sqrt(dx/msftx(i,j)*dy/msfty(i,j))
       pu1=pu(delxy,hpbl(i,j))
-      IF(sfcflg(i,j).and.sflux(i,j).GT.0.0)THEN
+      IF((sfcflg(i,j).and.sflux(i,j).GT.0.0) .and. (hpbl(i,j) .GT. 0)) THEN
```

Comment:
If the PBL height cannot be determined, e.g. because it is above the model top, hpbl = 0. This leads to a divide-by-zero error below.
The question is, whether, as in this fix, all further computations should be skipped for this grid point (pblflg(i,j) = .false.) resulting in a zero non-local flux or if there is a different solution.

Reply:
If hpbl=0, there does not exist nonlocal flux. I agree with your fix.